src/core.rs

```diff
@@ -431,9 +434,370 @@ impl Projectm {
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn set_window_size(&self, width: usize, height: usize) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::set_window_size(*instance, width, height);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn render_frame(&self) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::render_frame(*instance);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn touch(&self, x: f32, y: f32, pressure: i32, touch_type: ProjectMTouchType) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::touch(*instance, x, y, pressure, touch_type);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn touch_drag(&self, x: f32, y: f32, pressure: i32) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::touch_drag(*instance, x, y, pressure);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn touch_destroy(&self, x: f32, y: f32) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::touch_destroy(*instance, x, y);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn touch_destroy_all(&self) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::touch_destroy_all(*instance);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn pcm_get_max_samples() -> u32 {
+        Projectm::pcm_get_max_samples()
+    }
+
+    pub fn pcm_add_float(&self, samples: Vec<f32>, channels: ProjectMChannels) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::pcm_add_float(*instance, samples, channels);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn pcm_add_int16(&self, samples: Vec<i16>, channels: ProjectMChannels) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::pcm_add_int16(*instance, samples, channels);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn pcm_add_uint8(&self, samples: Vec<u8>, channels: ProjectMChannels) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::pcm_add_uint8(*instance, samples, channels);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
+    pub fn write_debug_image_on_next_frame(&self, output_file: Option<&String>) {
+        if let Ok(instance) = self.instance.try_borrow() {
+            Projectm::write_debug_image_on_next_frame(*instance, output_file);
+        } else {
+            panic!("Failed to borrow instance");
+        }
+    }
+
```

Comment:
Afaict, a `ProjectMHandle` instance is not `send` itself, but `ProjectM` is made to be both send and sync.

If we create an instance of `ProjectM` on one thread and then send it to another thread, one could call `get_instance` on both threads. 

Am I missing something that would prevent two `Rc`'s to the same C lib instance on different threads?